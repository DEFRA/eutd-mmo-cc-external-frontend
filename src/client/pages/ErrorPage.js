import React from 'react';
import {H1, Main} from 'govuk-react';
import PageTitle from '../components/PageTitle';
import {
  catchCertificatesTitle,
  processingStatementTitle,
  storageDocumentTitle
} from '../helpers/journeyConfiguration';

import { withTranslation } from 'react-i18next';
import { connect } from 'react-redux';

import i18n from '../../../src/i18n'
 export class ErrorPage extends React.Component {

  constructor(props){
    super(props)
  }
  state = {
    href: '/'
  };

  componentDidMount() {
    this.setState( {href: window.location.href});
    window.scrollTo(0, 0);
  }

  render() {
    const {href} = this.state;
    const {t} = i18n;
    let service = '';
    if( href.indexOf( '/create-processing-statement' ) !== -1) service = t(processingStatementTitle);
    if( href.indexOf( '/create-storage-document' ) !== -1) service = t(storageDocumentTitle);
    if( href.indexOf( '/create-catch-certificate' ) !== -1) service = t(catchCertificatesTitle);

    if( service ) service = `- ${service}`;
    else service = '- GOV.UK';


    return(
      <Main>
        <div className="column-two-thirds">
          <PageTitle title={`${t('commonErrorPageTitle')} ${service}`} />
          <H1>{t('commonErrorPageTitle')}</H1>
          <p>{t('commonErrorPageTryagainText')}</p>
          <p>{t('commonErrorPagesaveText')}</p>
        </div>
      </Main>
    );
  }
}

export const component = connect()(withTranslation() (ErrorPage))

export default {
  component: ErrorPage
};
