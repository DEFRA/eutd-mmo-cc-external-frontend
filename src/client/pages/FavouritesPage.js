import React from 'react';
import {
  Main,
  Header,
  BackLink,
  GridRow,
  GridCol,
  Table
} from 'govuk-react';
import PageTitle from '../components/PageTitle';
import FavouritesDetails from '../components/elements/FavouritesDetails';
import { connect } from 'react-redux';
import PageTemplateWrapper from '../components/PageTemplateWrapper';
import {
  dispatchApiCallFailed,
  getStatesFromReferenceService,
  getPresentationsFromReferenceService,
  getCommodityCode,
  searchFishStates,
  speciesSelectedFromSearchResult,
  searchUkFish,
  getAllUkFish,
  dispatchClearErrors
} from '../actions';
import { addFavourite, getAddedFavouritesPerUser, removeFavourite } from '../actions/favourites.actions';
import SpeciesBlock from '../components/SpeciesBlock';
import ErrorIsland from '../components/elements/ErrorIsland';
import { withTranslation } from 'react-i18next';
import { isEmpty } from 'lodash';
import SecondaryButton from '../components/elements/SecondaryButton';
import errorTransformer from '../helpers/errorTransformer';

export class FavouritesPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      favouriteId: undefined,
    };
    this.favouriteFormRef = null;

    this.setFavouriteFormRef = (element) => {
      this.favouriteFormRef = element;
    };
  }

  async componentDidMount() {
    try {
      await this.props.getAddedFavouritesPerUser();
    } catch (e) {
      console.error(e);
    }
  }

  unauthorised() {
    this.props.history.push('/forbidden');
  }

  componentDidUpdate() {
    const {
      unauthorised,
    } = this.props;

    if (
      unauthorised === true
    ) {
      this.unauthorised();
    }
  }

  goBackHandler = (event)=>{
    event.preventDefault();
    this.props.history.goBack();
  }

  removeFavouriteHandler = (event) => {
    const productId = event.currentTarget.id;
    this.props.removeFavourite(productId);
    this.setState({ favouriteId: undefined });
  };

  splitErrorParams = (error) => {
    const { t } = this.props;
    const isErrorWithParams = error.text.includes('-');
    const splitError = error.text.split('-');

    return isErrorWithParams ? t(splitError[0], { dynamicValue: splitError[1] }) : t(error.text);
  };

  formatErrors = (errors) => {
    return errors.map((error) => (
      { message: this.splitErrorParams(error), key: error.targetName }
    )
    );
  };

  render() {
    const { addedFavouritesPerUser = [], t } = this.props;
    const errors = errorTransformer(this.props.errors);

    return (
      <Main>
        <div className="favourites">
          <PageTitle title={t('ccFavouritesPageHeader')} />
          {!isEmpty(errors) && errors.errors && errors.errors.length > 0 && (
            <ErrorIsland
              errors={this.formatErrors(errors.errors)}
            />
          )}
          <GridRow>
            <GridCol>
              <BackLink
                onClick={(event) => this.goBackHandler(event)} href="#"
              >{t('commonBackLinkBackButtonLabel')}
              </BackLink>
              <Header>{t('ccFavouritesPageHeader')}</Header>
            </GridCol>
          </GridRow>
          <GridRow>
            <GridCol>
            <FavouritesDetails />
        <SpeciesBlock
          primaryButtonLabel={t('ccFavouritesPageFormCommodityCodeAddProductFavourites')}
          onSubmitAction={'/orchestration/api/v1/favourites/add'}
          errors={errors}
          formRef={this.setFavouriteFormRef}
          products={addedFavouritesPerUser || []}
          productId={this.state.favouriteId}
          speciesStates={this.props.speciesStates}
          speciesPresentations={this.props.speciesPresentations}
          getCommodityCode={this.props.getCommodityCode}
          commodityCodes={this.props.commodityCodes}
          searchFishStates={this.props.searchFishStates}
          fishStates={this.props.fishStates}
          displayAddProduct={true}
          cancelEditing={() => null}
          addFavourite={this.props.addFavourite}
          clearErrors={this.props.dispatchClearErrors}
        />
          <h2 className="heading-large" id="productTableCaption">
          {t('ccFavouritesPageProductTableCaption')}
        </h2>
        <GridRow>
          <GridCol>
            <Table
              id='product-favourites'
              head={(
                <Table.Row>
                  <Table.CellHeader>{t('ccFavouritesPageProductTableHeaderOne')}</Table.CellHeader>
                  <Table.CellHeader>{t('ccFavouritesPageProductTableHeaderTwo')} </Table.CellHeader>
                  <Table.CellHeader style={{ textAlign: 'right' }}>{t('ccFavouritesPageProductTableHeaderThree')}</Table.CellHeader>
                </Table.Row>
              )}
              body={(
                addedFavouritesPerUser.map((product, index) => (
                  <Table.Row id={`productfavourites_${product.id}`} key={index}>
                    <Table.Cell style={{ verticalAlign: 'top', width: '20%' }}>{product.id}</Table.Cell>
                    <Table.Cell style={{ verticalAlign: 'top', width: '50%'  }}><strong>{t('ccFavouritesPageProductTableRowSpecies')}</strong>: {product.species} <br/>
                                                                                <strong>{t('ccFavouritesPageProductTableRowState')}</strong>: {product.stateLabel} <br/>
                                                                                <strong>{t('ccFavouritesPageProductTableRowPresentation')}</strong>: {product.presentationLabel} <br/>
                                                                                <strong>{t('ccFavouritesPageProductTableRowCommodityCode')}</strong>: {`${product.commodity_code} ${product.commodity_code_description ? '- '  + product.commodity_code_description : ''}`.trim()}
                                                                               </Table.Cell>
                    <Table.Cell style={{ verticalAlign: 'top', width: '10%'  }}>
                    <GridRow style={{ margin: 0, placeContent: 'flex-end' }}>
                    <SecondaryButton
                        type="button"
                        id={product.id}
                        name="removeProduct"
                        style={{ minWidth: '100px' }}
                        onClick={this.removeFavouriteHandler}
                      >
                        {t('ccFavouritesPageProductTableRowRemoveButton')}
                        <span className="govuk-visually-hidden">
                          {`productfavourites_${product.id}`}
                        </span>
                      </SecondaryButton>
                      </GridRow>
                    </Table.Cell>
                  </Table.Row>
               ))
            )}
            />
          </GridCol>
        </GridRow>
       </GridCol>
          </GridRow>
        </div>
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    exporter: state.exporter,
    addedFavouritesPerUser: state.addedFavouritesPerUser.favourites,
    addedSpeciesErrorState: state.addedSpeciesErrorState,
    speciesStates: state.speciesStates,
    unauthorised: state.addedFavouritesPerUser.unauthorised,
    speciesPresentations: state.speciesPresentations,
    commodityCodes: state.commodityCodes,
    fishStates: state.fishStates,
    errors: state.addedFavouritesPerUser.errors,
    searchResultsForSpecies: state.fish,
    config: state.config,
  };
}

async function loadData(store) {
  const hasErrorParams =
    this.queryParams &&
    this.queryParams.error &&
    Object.keys(this.queryParams.error).length;
  if (hasErrorParams) {
    store.dispatch(dispatchApiCallFailed(JSON.parse(this.queryParams.error)));
  }

  await Promise.all([
    store.dispatch(getStatesFromReferenceService()),
    store.dispatch(getAddedFavouritesPerUser),
    store.dispatch(getPresentationsFromReferenceService()),
    store.dispatch(getAllUkFish())
  ]);
}

export const component = connect(mapStateToProps, {
  getStatesFromReferenceService,
  getPresentationsFromReferenceService,
  getCommodityCode,
  searchFishStates,
  speciesSelectedFromSearchResult,
  searchUkFish,
  getAllUkFish,
  addFavourite,
  removeFavourite,
  dispatchClearErrors,
  getAddedFavouritesPerUser,
})(withTranslation() (FavouritesPage));

export default {
  loadData,
  component: PageTemplateWrapper(component),
};