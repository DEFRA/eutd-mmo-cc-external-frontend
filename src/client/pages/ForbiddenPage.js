import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import PageTitle from '../components/PageTitle';
import i18n from '../../../src/i18n';

export class ForbiddenPage extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount(){
    i18n.changeLanguage();
    this.forceUpdate();
  }

  render() {
    const {t} = i18n;
    const { reference, exporter, exportPayload, transport, processingStatement, storageNotes, directLandings, conservation } = this.props;
    const supportID =
      reference.supportID ||
      exporter.supportID ||
      exportPayload.supportID ||
      transport.supportID ||
      processingStatement.supportID ||
      storageNotes.supportID ||
      directLandings.supportID ||
      conservation.supportID;

    return(
        <div className="column-two-thirds">
          <PageTitle title={`${t('commonForbiddenPageText')} â€“ GOV.UK`} />
          {supportID 
            ? <>
              <h1 className="heading-large">{t('commonForbiddenPageURLRejected')}</h1>
              <p data-testid="try-again">{t('commonForbiddenPageTryAgain')}</p>
              <p data-testid="support-id">{t('commonForbiddenPageSupportId')} {supportID}</p>
              <p data-testid="contact-fes">{t('commonForbiddenPageContact')}</p>
              <h2 className="heading-large" data-testid="subheading-fes">{t('commonFishExportsService')}</h2>
              <p data-testid="tel-number">{t('commonTelephone')}: 0330 159 1989</p>
              <p data-testid="opening-hours">{t('commonForbiddenPageOpeningHours')}</p>
            </>
            : <>
              <h1 className="heading-large">{t('commonForbiddenPageText')}</h1>
              <p data-testid="no-permission">{t('commonForbiddenPageP1Text')}</p>
              <p data-testid="navigate-back">{t('commonForbiddenPageP2Text')}</p>
            </>
          }
        </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    reference: state.reference,
    exporter: state.exporter,
    exportPayload: state.exportPayload,
    transport: state.transport,
    processingStatement: state.processingStatement,
    storageNotes: state.storageNotes,
    directLandings: state.directLandings,
    conservation: state.conservation
  };
}

export default {
  component: withRouter(connect(mapStateToProps, {})(ForbiddenPage)),
};
