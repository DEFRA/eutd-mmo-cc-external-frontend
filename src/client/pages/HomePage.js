import React from 'react';
import { connect } from 'react-redux';
import { Header, Main, Button } from 'govuk-react';
import SelectRadio from '../../../src/client/components/elements/SelectRadio';
import PageTitle from '../components/PageTitle';
import Form from '../components/elements/Form';
import { withTranslation } from 'react-i18next';
import {withRouter} from 'react-router-dom';
import i18n from '../../i18n';
class Home extends React.Component {

  state = {
    selected: '/create-catch-certificate/catch-certificates',
  };

  componentDidMount(){
    i18n.changeLanguage();
    this.forceUpdate();
  }

  render() {
    const {  t } = this.props;
    return (
      <Main>
        <PageTitle title= {`${t('commonHomePageWhatDoYouWant')}`} />
        <Header>{t('commonHomePageWhatDoYouWant')}</Header>
        <Form
          currentUrl='/'
          action="/orchestration/api/v1/home/saveAndValidate"
        >
          <div id="radioButtons">
            <fieldset className="govuk-fieldset">
              <legend className="visually-hidden">{t('commonHomePageWhatDoYouWant')}</legend>
              <SelectRadio onClick={() => this.setState({ selected: '/create-catch-certificate/catch-certificates' })} id="createCatchCertificate" name="journeySelection" value="/create-catch-certificate/catch-certificates" defaultChecked>{t('commonDashboardCreateAUkCatchCertificate')} {t('commonHomePageIncludingLinksText')}</SelectRadio>
              <SelectRadio onClick={() => this.setState({selected: '/create-processing-statement/processing-statements'})} id="createProcessingStatement" name="journeySelection" value="/create-processing-statement/processing-statements">{t('processingStatementRendererHeaderSectionTitle')}</SelectRadio>
              <SelectRadio onClick={() => this.setState({selected: '/create-storage-document/storage-documents'})} id="createStorageDocument" name="journeySelection" value="/create-storage-document/storage-documents">{t('commonDashboardCreateaUkStorageDocument')}</SelectRadio>
            </fieldset>
          </div>
          <br/>
          <Button id="continue" name="continue" type="submit">{t('commonContinueButtonContinueButtonText')}</Button>
        </Form>
      </Main>
    );
  }
}

export default {
  component: withRouter(connect(null, {})(withTranslation() (Home)))
};
