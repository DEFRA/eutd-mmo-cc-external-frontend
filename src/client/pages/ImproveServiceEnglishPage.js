import React from 'react';
import {
  Main,
  BackLink,
  GridRow,
  GridCol,
  Header,
  UnorderedList,
  ListItem,
} from 'govuk-react';
import PageTitle from '../components/PageTitle';

const ImproveServiceEnglishPage = ({
  feedbackUrl,
  listFont,
  backRoute,
  goBack,
  title,
}) => (
  <Main>
    <PageTitle title={`Service improvement plan - ${title}`} />
    <BackLink
      onClick={(e) => {
        e.preventDefault();
        goBack();
      }}
      href={backRoute}
    />
    <GridRow>
      <GridCol>
        <Header level="1">How we're improving the service</Header>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol>
        <p>
          The <strong>Fish Exports Service</strong> helps businesses operating
          in the UK produce documents that enable fish exports to the EU and
          certain other countries.
        </p>
        <p>The service:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem style={listFont}>
            uses your information in a secure way
          </ListItem>
          <ListItem style={listFont}>
            is working towards the WCAG 2.1 AA accessibility standard
          </ListItem>
          <ListItem style={listFont}>
            uses technology that allows it to be improved over time
          </ListItem>
          <ListItem style={listFont}>
            aims to meet the Governmentâ€™s{' '}
            <a
              href="https://www.gov.uk/service-manual/service-standard"
              rel="noopener noreferrer"
              target="_blank"
              alt="Opens service manual"
            >
              Digital Service Standard (opens in new tab)
            </a>
            .
          </ListItem>
        </UnorderedList>
        <br />
        <p>As part of continuous improvement, the service aims to:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem style={listFont}>
            address and fix identified, WCAG 2.1 AA non-compliant accessibility
            issues (priority 1)
          </ListItem>
          <ListItem style={listFont}>
            carry out continuous research to make sure you can complete the
            service first time, and make improvements where necessary (priority
            2)
          </ListItem>
          <ListItem style={listFont}>
            continue to research and test the service to ensure your needs are
            addressed (priority 3)
          </ListItem>
        </UnorderedList>
        <br />
        <p>
          Your{' '}
          <a
            href={feedbackUrl}
            rel="noopener noreferrer"
            target="_blank"
            alt="Opens feedback form"
          >
            feedback (opens in new tab)
          </a>{' '}
          will help us to improve this service.
        </p>
      </GridCol>
    </GridRow>
  </Main>
);

export default ImproveServiceEnglishPage;
