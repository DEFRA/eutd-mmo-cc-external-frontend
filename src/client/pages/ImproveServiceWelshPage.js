import React from 'react';
import {
  Main,
  BackLink,
  GridRow,
  GridCol,
  Header,
  UnorderedList,
  ListItem,
} from 'govuk-react';
import PageTitle from '../components/PageTitle';

const ImproveServiceWelshPage = ({
  feedbackUrl,
  listFont,
  backRoute,
  goBack,
  title,
}) => (
  <Main>
    <PageTitle title={`Cynllun gwella gwasanaeth - ${title}`} />
    <BackLink
      onClick={(e) => {
        e.preventDefault();
        goBack();
      }}
      href={backRoute}
    >
      Yn ôl
    </BackLink>
    <GridRow>
      <GridCol>
        <Header level="1">Sut rydyn ni’n gwella’r gwasanaeth</Header>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol>
        <p>
          Mae’r <strong>Gwasanaeth Allforio Pysgod</strong> yn helpu busnesau
          sy’n gweithredu yn y DU i gynhyrchu dogfennau sy’n golygu bod modd
          allforio pysgod i’r UE ac i rai gwledydd eraill.
        </p>
        <p>Mae’r gwasanaeth yn:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem style={listFont}>
            defnyddio eich gwybodaeth mewn ffordd ddiogel
          </ListItem>
          <ListItem style={listFont}>
            gweithio tuag at safon hygyrchedd WCAG 2.1 AA
          </ListItem>
          <ListItem style={listFont}>
            defnyddio technoleg sy’n ei alluogi i wella dros amser
          </ListItem>
          <ListItem style={listFont}>
            Anelu at gyrraedd{' '}
            <a
              href="https://www.gov.uk/service-manual/service-standard"
              rel="noopener noreferrer"
              target="_blank"
              alt="Opens service manual"
            >
              Safon Gwasanaeth Digidol y Llywodraeth (yn agor mewn tab newydd)
            </a>
            .
          </ListItem>
        </UnorderedList>
        <br />
        <p>Er mwyn gwella’n barhaus, nod y gwasanaeth yw:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem style={listFont}>
            canfod a datrys problemau hygyrchedd sydd ddim yn cydymffurfio â
            WCAG 2.1 AA (blaenoriaeth 1)
          </ListItem>
          <ListItem style={listFont}>
            gwneud gwaith ymchwil parhaus i wneud yn siŵr eich bod yn gallu
            cwblhau’r gwasanaeth y tro cyntaf, a gwneud gwelliannau lle bo angen
            (blaenoriaeth 2)
          </ListItem>
          <ListItem style={listFont}>
            parhau i ymchwilio a phrofi’r gwasanaeth fel ei fod yn cyflawni eich
            anghenion (blaenoriaeth 3)
          </ListItem>
        </UnorderedList>
        <br />
        <p>
          Bydd eich{' '}
          <a
            href={feedbackUrl}
            rel="noopener noreferrer"
            target="_blank"
            alt="Opens feedback form"
          >
            adborth (yn agor mewn tab newydd)
          </a>{' '}
          yn ein helpu i wella’r gwasanaeth hwn.
        </p>
      </GridCol>
    </GridRow>
  </Main>
);

export default ImproveServiceWelshPage;
