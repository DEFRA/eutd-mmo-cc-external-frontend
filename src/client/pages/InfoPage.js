import React from 'react';
import { connect } from 'react-redux';
import { Main, H3 } from 'govuk-react';
import {getVersionInfo, getReferenceDataReaderVersionInfo} from '../actions';

class InfoPage extends React.Component {

  async componentDidMount() {
    await this.props.getVersionInfo();
    await this.props.getReferenceDataReaderVersionInfo();
  }

  render() {

    const {orchestrationVersionInfo={}, referenceDataReaderVersionInfo={}} = this.props.global;
    const {config} = this.props;

    return (
      <Main>
        <H3>mmo-ecc-fe</H3>
        <p id="mmo-ecc-fe">
          Version: {config.versionInfo.gitHash}
        </p>
        <hr/>
        <H3>mmo-ecc-orchestration</H3>
        <p id="mmo-ecc-orchestration">
          Version: {orchestrationVersionInfo.gitHash}
        </p>
        <hr/>
        <H3>mmo-ecc-reference-data-reader</H3>
        <p id="mmo-ecc-reference-data-reader">
          Version: {referenceDataReaderVersionInfo.gitHash}
        </p>
      </Main>
    );
  }
}

export const component = connect( (state) => ({config: state.config, global: state.global}), {getVersionInfo, getReferenceDataReaderVersionInfo} )(InfoPage);

export default {
  component
};
