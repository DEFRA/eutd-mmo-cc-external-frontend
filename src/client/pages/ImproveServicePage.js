import React, { Component } from 'react';
import { connect } from 'react-redux';
import i18n from '../../../src/i18n';
import { getAllUserAttributes } from '../actions/userAttributes.actions';
import PageTemplateWrapper from '../components/PageTemplateWrapper';
import ImproveServiceWelshPage from './ImproveServiceWelshPage';
import ImproveServiceEnglishPage from './ImproveServiceEnglishPage';

function getService(backRoute) {
  if( backRoute.indexOf( '/create-processing-statement' ) !== -1) return {title: i18n.t('psCommonTitle'), href: '/create-processing-statement/processing-statements' };
  if( backRoute.indexOf( '/create-storage-document' ) !== -1) return {title: i18n.t('sdCommonTitle'), href: '/create-storage-document/storage-documents' };
  if( backRoute.indexOf( '/create-catch-certificate' ) !== -1) return {title: i18n.t('ccCommonTitle'), href: '/create-catch-certificate/catch-certificates' };
  return {title: '', href: '#'};
}
class ImproveServicePage extends Component {

  state = {
    backRoute: '#'
  };

  componentDidMount() {
    const backRoute = window.location.search.replace('?', '');
    this.setState( {backRoute});
    const service = getService(backRoute);
    const titleNode = document.getElementsByClassName('govuk-header__link--service-name');
    if( titleNode && titleNode[0] ) {
      titleNode[0].text = service.title.replace(' - GOV.UK', '');
      titleNode[0].href = service.href;
    }
  }

  getSelectedLanguage = (userAttributes) => {
    const userLanguageAttribute = userAttributes.find(
      (userAttribute) => userAttribute.name === 'language'
    );
    if (userLanguageAttribute) {
      return userLanguageAttribute.value;
    }
  };

  render() {
    const { backRoute } = this.state;
    const {feedbackUrl, userAttributes } = this.props
    const userSelectedLanguage = this.getSelectedLanguage(userAttributes);

    let title = getService(backRoute).title;
    if( !title ) title = 'GOV.UK';

    const listFont = {
      'fontSize': '19px'
    };

    return userSelectedLanguage === 'cy_UK' ? (
      <ImproveServiceWelshPage
        feedbackUrl={feedbackUrl}
        backRoute={backRoute}
        listFont={listFont}
        goBack={() => this.props.history.goBack()}
        title={title}
      />
    ) : (
      <ImproveServiceEnglishPage
        feedbackUrl={feedbackUrl}
        backRoute={backRoute}
        listFont={listFont}
        goBack={() => this.props.history.goBack()}
        title={title}
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    feedbackUrl: state.config.feedbackUrl,
    userAttributes: state.userAttributes,
  };
}

function loadData(store) {
  return Promise.all([store.dispatch(getAllUserAttributes())]);
}

export const component = connect(mapStateToProps, {
  getAllUserAttributes,
})(ImproveServicePage);

export default {
  loadData,
  component: PageTemplateWrapper(component),
};
