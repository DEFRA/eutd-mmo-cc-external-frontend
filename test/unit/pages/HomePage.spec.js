import React from 'react';
import { mount } from 'enzyme';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import * as HomePage from '../../../src/client/pages/HomePage';
import { render } from '@testing-library/react';
jest.mock('../../../src/client/actions/userAttributes.actions');

describe('Home Page', () => {
  let wrapper;

  const mockStore = configureStore([thunk]);

  beforeAll(() => {
    const store = mockStore({});
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <HomePage.default.component />
        </MemoryRouter>
      </Provider>
    );
  });

  it('should load successfully', () => {
    expect(wrapper).toBeDefined();
  });

  it('should create snapshot for whole page',() => {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  })

  it('should render radio button', () => {
    expect(wrapper.find({ type: 'radio' })).toBeDefined();
  });

  it('should handle an submit event', () => {
    wrapper.setState({
      selected: '/create-catch-certificate/catch-certificates',
    });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
  });

  it('should handle radio button click event', () => {
    wrapper.find('input#createCatchCertificate').simulate('click');
    wrapper.find('input#createProcessingStatement').simulate('click');
    wrapper.find('input#createStorageDocument').simulate('click');
  });

  it('should handle continue button click event', () => {
    wrapper.setState({
      selected: '/create-catch-certificate/catch-certificates',
    });
    wrapper.find('button#continue').simulate('click');
  });

  it('should handle a submit event', () => {
    wrapper.setState({
      selected: '/create-catch-certificate/catch-certificates',
    });
    wrapper
      .find('button#continue')
      .simulate('click', { preventDefault() {} });
  });

  it('should have a for attribute on all input labels', () => {
    expect(
      wrapper.find('label#label-createCatchCertificate').props()['htmlFor']
    ).toBeDefined();
    expect(
      wrapper.find('label#label-createCatchCertificate').props()['htmlFor']
    ).toBe('createCatchCertificate');
    expect(
      wrapper.find('label#label-createProcessingStatement').props()['htmlFor']
    ).toBeDefined();
    expect(
      wrapper.find('label#label-createProcessingStatement').props()['htmlFor']
    ).toBe('createProcessingStatement');
    expect(
      wrapper.find('label#label-createStorageDocument').props()['htmlFor']
    ).toBeDefined();
    expect(
      wrapper.find('label#label-createStorageDocument').props()['htmlFor']
    ).toBe('createStorageDocument');
  });
});
