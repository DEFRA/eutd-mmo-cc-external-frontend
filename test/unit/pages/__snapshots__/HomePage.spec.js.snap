// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should create snapshot for whole page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <h1
        class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
      >
        What do you want to do?
      </h1>
      <form
        action="/orchestration/api/v1/home/saveAndValidate?c=/"
        method="POST"
      >
        <div
          id="radioButtons"
        >
          <fieldset
            class="govuk-fieldset"
          >
            <legend
              class="visually-hidden"
            >
              What do you want to do?
            </legend>
            <label
              class="css-sdrcg6-StyledHoc"
              for="createCatchCertificate"
              id="label-createCatchCertificate"
            >
              <input
                checked=""
                class="css-1xt3i3l"
                id="createCatchCertificate"
                name="journeySelection"
                type="radio"
                value="/create-catch-certificate/catch-certificates"
              />
              <span
                class="css-1uwxoxa"
              >
                Create a UK catch certificate
                 
                (Including links to direct landing documents)
              </span>
            </label>
            <label
              class="css-sdrcg6-StyledHoc"
              for="createProcessingStatement"
              id="label-createProcessingStatement"
            >
              <input
                class="css-1xt3i3l"
                id="createProcessingStatement"
                name="journeySelection"
                type="radio"
                value="/create-processing-statement/processing-statements"
              />
              <span
                class="css-1uwxoxa"
              >
                Create a UK processing statement
              </span>
            </label>
            <label
              class="css-sdrcg6-StyledHoc"
              for="createStorageDocument"
              id="label-createStorageDocument"
            >
              <input
                class="css-1xt3i3l"
                id="createStorageDocument"
                name="journeySelection"
                type="radio"
                value="/create-storage-document/storage-documents"
              />
              <span
                class="css-1uwxoxa"
              >
                Create a UK storage document
              </span>
            </label>
          </fieldset>
        </div>
        <br />
        <button
          class="e1vcssft0 css-1enchvi-StyledButton-StyledHoc e4u1u2g0"
          id="continue"
          name="continue"
          type="submit"
        >
          Continue
        </button>
      </form>
    </div>
  </div>
</div>
`;
